compiler/types/Multiplicity.hs

```diff
@@ -76,12 +76,13 @@ isMultMul ty | Just (tc, [x, y]) <- splitTyConApp_maybe ty
 pattern MultMul :: Mult -> Mult -> Mult
 pattern MultMul p q <- (isMultMul -> Just (p,q))
 
+-- For now, approximate p * q by Omega unless p or q are known to be One.
 mkMultMul :: Mult -> Mult -> Mult
 mkMultMul One p = p
 mkMultMul p One = p
 mkMultMul Omega _ = Omega
 mkMultMul _ Omega = Omega
-mkMultMul p q = mkTyConApp multMulTyCon [p, q]
+mkMultMul _ _ = Omega
```

Comment:
We're basically dropping the support for polymorphism here. Which is fair, it doesn't really work. But this is a big XXX.